complete-firebase-authentication-react-tutorial.md

```diff
@@ -1,365 +1,353 @@
 
 {{% chapter_header "Table of Contents" "toc" %}}
 
 * [React Application Setup: create-react-app](#react-application-setup)
-* [React Router and Routes](#react-router-setup)
+* [React Router for Firebase Auth](#react-router-setup)
+* Firebase Setup
+  * [Firebase in React](#react-firebase)
+  * [Provide Firebase in React](#react-firebase-context-api)
 * Firebase Authentication in React
-  * [Firebase in React Setup](#react-firebase-setup)
   * [Firebase's Authentication API](#firebase-authentication)
   * [Sign Up with React and Firebase](#react-firebase-sign-up)
   * [Sign In with React and Firebase](#react-firebase-sign-in)
   * [Sign Out with React and Firebase](#react-firebase-sign-out)
-  * [Session Handling in React Components](#react-firebase-session-handling)
-  * [Session Handling in React with Higher Order Components](#react-firebase-session-handling-higher-order-components)
+  * [Session Handling with Firebase/React](#react-firebase-session-handling)
+  * [Session Handling with Higher-Order Components](#react-firebase-session-handling-higher-order-components)
   * [Password Reset and Password Change with Firebase](#react-firebase-password)
-* Authorization in React
+* Firebase Authorization in React
   * [Protected Routes in React with Authorization](#react-firebase-protected-routes)
-* Firebase Database in React
-  * [User Management with Firebase's Database in React](#react-firebase-user-database)
+* Firebase Realtime Database in React
+  * [Firebase Realtime Database in React](#react-firebase-realtime-database)
+  * [Manage Users with Firebase's Realtime Database in React](#react-firebase-realtime-database-user-management)
 * [What's next?](#react-firebase-authentication-advanced)
-  * Bonus: [Authentication in React, Firebase and Redux](#react-firebase-authentication-redux)
-  * Bonus: [Authentication in React, Firebase and MobX](#react-firebase-authentication-mobx)
 
 {{% chapter_header "React Application Setup: create-react-app" "react-application-setup" %}}
 
-You are going to implement a whole authentication mechanism in React with sign up, sign in and sign out. Furthermore, it should be possible to reset a password or change a password as a user. The latter option is only available for an authenticated user. Last but not least, it should be possible to protect certain routes (URLs) to be accessible by authenticated users only. Therefore you will build a proper authorization solution around it.
+There are two ways to begin the lesson for this application: either follow my guidance in this section; or find a starter project in this {{% a_blank "GitHub repository" "hhttps://github.com/the-road-to-react-with-firebase/react-firebase-authentication-starter-kit" %}} and follow its installation instructions. [Editors Note: Tell them why they should choose one or the other, else you're suggesting an alternative route for no reason]
```

Comment:
Good point:

"[Editors Note: Tell them why they should choose one or the other, else you're suggesting an alternative route for no reason]" -> "Whereas the former will show you how to set up the whole project from scratch, the latter will give you instant access to a starter project where you don't need to worry about setting up the folder/file structure yourself."